creator-node/src/services/sync/secondarySyncFromPrimary.js

```diff
@@ -47,6 +61,8 @@ const handleSyncFromPrimary = async (
         ),
         result: 'failure_sync_in_progress'
       }
+    } finally {
+      await releaseRedisLocks(walletPublicKeys, redis, logPrefix)
```

Comment:
we don't want this here. that means another sync has the lock and we should not clear the lock the other sync here

Reply:
ah I see. I was thinking if wallet N fails to acquire then we'd want to release the locks for wallets [0...N-1], but we only ever sync with an array of size 1 if I understand correctly. I'll remove this

Reply:
the size of the wallets shouldn't matter here. for example if we have wallets [0x01, 0x02] in a sync and we have two secondarySyncFromPrimary functions running

function 1 - lock acquired for 0x01 and 0x02
function 2- trying to acquire lock for 0x01. will fail so the code you've added would clear lock process 1 has acquired. same for process 2

so this might be a problem if we had 1 wallet or n wallets. but we just so happen to only sync one wallet at a time

Reply:
ah gotcha, that makes sense if the array is in the same order each time. since there's only ever 1 wallet this isn't important, but the scenario I was picturing this fixing was:

process 1 - lock acquired for 0x02 and 0x03
process 2 - lock acquired for 0x01 and failed for 0x02 then it would be best to clear 0x01 now rather than waiting for it to time out since that lock was only acquired by this process and this process is short circuiting